lib/src/configuration.dart

```diff
@@ -216,24 +47,40 @@ class Configuration {
   /// Realm with a schema that contains objects that differ from their previous 
   /// specification. If the schema was updated and the schemaVersion was not, 
   /// an [RealmException] will be thrown.
-  double? schemaVersion;
+  int get schemaVersion => realmCore.getSchemaVersion(this);
+  set schemaVersion(int value) => realmCore.setSchemaVersion(this, value);
+
+  String get path => realmCore.getConfigPath(this);
+  set path(String value) => realmCore.setConfigPath(this, value);
+}
+
+class SchemaObject {
+  Type type;
+  String get name => type.toString();
 
-  /// Specifies if this Realm should be deleted if a migration is needed.
-  bool? deleteRealmIfMigrationNeeded;
+  List<SchemaProperty> properties = [];
+
+  SchemaObject(this.type);
+}
+
+class RealmSchema extends Iterable<SchemaObject> {
+  late final SchemaHandle handle;
```

Comment:
I'm new to dart, but should this (and other such handles) be public? Sounds like users should never need to use them.

Reply:
yes handles need to be public but they are kept with empty signatures (except toString()) on purpose so nobody have any useful thing to do with them.

Reply:
Why do they need to be public though? Is it an ffi limitation or something that the generator needs to use?

Reply:
In Dart private members are private to the library not the package. What issue do you see with these being public except that these are one extra field on certain Realm classes? They do not provide anything useful and couldn't be used in any manner whatsoever.

Reply:
@nirinchev Can I resolve this comment

Reply:
resolving

Reply:
@nirinchev we have moved these handles to private here https://github.com/realm/realm-dart/blob/master/lib/src/configuration.dart#L32